tests/models/test_selector.py

```diff
@@ -0,0 +1,113 @@
+"""Tests for the models selector"""
+
+import json
+
+import pytest
+
+from ralph.models.edx.browser import PageCloseBrowserEventModel
+from ralph.models.edx.server import ServerEventModel
+from ralph.models.selector import LazyModelField, ModelSelector, Rule
+
+from tests.fixtures.edx.browser import PageCloseBrowserEventFactory
+from tests.fixtures.edx.server import ServerEventFactory
+
+
+@pytest.mark.parametrize(
+    "model_rules,decision_tree",
+    [
+        ({}, {}),
+        (
+            {ServerEventModel: [Rule("event_source", "server")]},
+            {
+                Rule("event_source", "server"): {
+                    True: ServerEventModel,
+                    False: None,
+                }
+            },
+        ),
+        (
+            {
+                ServerEventModel: [
+                    Rule("event_source", "server"),
+                    Rule("event_type", LazyModelField("context__path")),
+                ]
+            },
+            {
+                Rule("event_source", "server"): {
+                    True: {
+                        Rule("event_type", LazyModelField("context__path")): {
+                            True: ServerEventModel,
+                            False: None,
+                        },
+                    },
+                    False: None,
+                }
+            },
+        ),
+        (
+            {
+                PageCloseBrowserEventModel: [
+                    Rule("event_source", "browser"),
+                    Rule("event_type", "page_close"),
+                ],
+                ServerEventModel: [
+                    Rule("event_source", "server"),
+                    Rule("event_type", LazyModelField("context__path")),
+                ],
```

Comment:
I think you should add more complex scenario with similar events using common rules.

Reply:
Good Idea) Thanks) I have added one more test with 3 models, multiple rules, where one rule occurs twice.